### DISCUSSION TITLE
Netbox plugin interface

### DISCUSSION THREAD

**Discussion by peteeckel** [auroraresearchlab/netbox-dns#123] [General]
Some good news from the upstream project.

With the last few Netbox releases we had some trouble with incompatible changes in some Netbox APIs we are using, but strictly speaking shouldn't use because they are not officially supported and subject to sudden changes.

Obviously we haven't been the only ones who were bitten by that, because the temptation and, in some cases, the necessity to use those APIs is overwhelming when a reasonable feature set is the goal. And the Netbox maintainers reacted by setting up a workgroup with the aim of figuring out what the new plugin interface should look like. IMHO this looks very promising even at this early stage.

The very good news is that they are actually listening (the initial discussion is [here](https://github.com/netbox-community/netbox/discussions/8338), and the results that have already led to new developments in Netbox alone will help us a lot with getting a cleaner and more stable interface.

So for the time being I've postponed the development of the 'views' functionality in the hope that Netbox 3.2 (which is the scheduled version for the beefed-up plugin interface) will make many things easier and more stable. Fingers crossed!

**Comment by jeremystretch** [comment]
:wave: Hey all! For anyone interested, over on the NetBox side we're tracking the overall plugins framework improvement initiative under netbox-community/netbox#8333. We have a lot of improvements already underway for v3.2, and I expect to continue extending the framework over successive releases.

Just wanted to chime in to thank @peteeckel for volunteering to help with this effort, and to note that this is one of the plugins I'm going to use as a guinea pig for testing under v3.2. I hope to be able to provide some assistance in adapting for v3.2 in the near future.
